packages/react-instantsearch-core/src/core/createInstantSearchManager.js

```diff
@@ -3,6 +3,15 @@ import createWidgetsManager from './createWidgetsManager';
 import createStore from './createStore';
 import { HIGHLIGHT_TAGS } from './highlight';
 import { hasMultipleIndices } from './indexUtils';
+import { version as ReactVersion } from 'react';
+import version from './version';
+
+export function addAlgoliaAgents(searchClient) {
+  if (typeof searchClient.addAlgoliaAgent === 'function') {
+    searchClient.addAlgoliaAgent(`react (${ReactVersion})`);
+    searchClient.addAlgoliaAgent(`react-instantsearch (${version})`);
```

Comment:
We should be able to add additional agent from the environment, DOM, Native, Server.